ssz-rs/src/merkleization/mod.rs

```diff
@@ -44,9 +44,11 @@ impl Display for MerkleizationError {
 impl std::error::Error for MerkleizationError {}
 
 pub fn pack_bytes(buffer: &mut Vec<u8>) {
-    let data_len = buffer.len();
-    if data_len % BYTES_PER_CHUNK != 0 {
-        let bytes_to_pad = BYTES_PER_CHUNK - data_len % BYTES_PER_CHUNK;
+    let incomplete_chunk_len = buffer.len() % BYTES_PER_CHUNK;
+    if incomplete_chunk_len != 0 {
+        // SAFETY: checked subtraction is unnecessary,
+        // as BYTES_PER_CHUNK > incomplete_chunk_len; qed
+        let bytes_to_pad = BYTES_PER_CHUNK - incomplete_chunk_len;
         let pad = vec![0u8; bytes_to_pad];
         buffer.extend_from_slice(&pad);
```

Comment:
we can skip the allocation (via `vec!`) here

```suggestion
        buffer.resize(buffer.len() + bytes_to_pad, 0);
```